src/Shared/UnitTests/TestEnvironment.cs

```diff
@@ -44,18 +46,38 @@ private TestEnvironment(ITestOutputHelper output)
             SetDefaultInvariant();
         }
 
+        public void Dispose()
+        {
+            GC.SuppressFinalize(this);
+            Cleanup();
+        }
+
+        ~TestEnvironment()
+        {
+            Cleanup();
+        }
+
         /// <summary>
         ///     Revert / cleanup variants and then assert invariants.
         /// </summary>
-        public void Dispose()
+        private void Cleanup()
```

Comment:
This is similar to but different from "the usual way": https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose

Is that intentional?

Reply:
Nope, I just did it "by hand" :). I think the behaviour is equivalent. Did another commit that moves the finalizer suppression at the end, just-in-case.